pkg/service/service.go

```diff
@@ -96,36 +96,27 @@ func Apply(
 ) (err error) {
 	logger := log.WithName("service")
 
-	defer func() {
-		if err == nil {
-			return
-		}
-		// Update a failure status.
-		status.ServiceStatusDeployed.False(entity, err.Error())
-
-		uerr := UpdateStatus(ctx, mc, entity)
-		if uerr != nil {
-			logger.Errorf("error updating status of service %s: %v",
-				entity.ID, uerr)
-		}
-	}()
-
 	if !status.ServiceStatusDeployed.IsUnknown(entity) {
 		return errorx.Errorf("service status is not deploying, service: %s", entity.ID)
 	}
 
-	if dp == nil {
-		return errorx.New("deployer is not set")
-	}
-
 	applyOpts := deptypes.ApplyOptions{
 		SkipTLSVerify: !opts.TlsCertified,
 		Tags:          opts.Tags,
 	}
 
 	err = dp.Apply(ctx, entity, applyOpts)
 	if err != nil {
-		return errorx.Errorf("failed to apply service: %v", err)
+		err = fmt.Errorf("failed to apply service: %v", err)
```

Comment:
use `%w` instead of `%v`, then could unwrap the original error while needed.

Reply:
Updated